### ISSUE TITLE
[JENKINS-3966] Add option to clean workspace before each build

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06#174]
<p>If this option is selected the workspace will be deleted before each build<br/>
begins so that each build can have a fresh workspace.  This will be off by default.</p>

---
<details><summary><i>Originally reported by <a title="kaxelson" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kaxelson">kaxelson</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-3966" target="_blank">Add option to clean workspace before each build</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="kaxelson" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kaxelson">kaxelson</a>
<li><b>reported by</b>: kaxelson
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2014-05-23T15:20:46+00:00
<li><b>votes</b>: 19
<li><b>watchers</b>: 21
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kaxelson">kaxelson</a>:</i>
<p>starting work on this</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in hudson<br/>
User: : kaxelson<br/>
Path:<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractBuild.java<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java<br/>
 trunk/hudson/main/core/src/main/resources/hudson/model/AbstractItem/configure-common.jelly<br/>
 trunk/hudson/main/core/src/main/resources/lib/hudson/project/config-cleanWorkspace.jelly<br/>
 trunk/hudson/main/war/resources/help/project-config/cleanWorkspace.html<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19373" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19373</a><br/>
Log:<br/>
<span class="error">&#91;FIXED JENKINS-3966&#93;</span> added option to clean workspace before each build as part of advanced project configuration</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in hudson<br/>
User: : kaxelson<br/>
Path:<br/>
 trunk/www/changelog.html<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19374" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19374</a><br/>
Log:<br/>
<a href="https://issues.jenkins.io/browse/JENKINS-3966" title="Add option to clean workspace before each build" class="issue-link" data-issue-key="JENKINS-3966"><del>JENKINS-3966</del></a>: adding changelog message</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>:</i>
<p>Rolled back from 1.315 release in rev.19544. See<br/>
<a href="http://www.nabble.com/attention-all-subversion-users-td24335693.html" class="external-link" target="_blank" rel="nofollow noopener">http://www.nabble.com/attention-all-subversion-users-td24335693.html</a> for the<br/>
discussion.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>:</i>
<p>Note that the Mercurial plugin already has an option to delete all nonversioned<br/>
files from the workspace before starting. (The .hg/ subdirectory is excluded as<br/>
is usual.) I think there is a similar issue open for Subversion. This is more<br/>
efficient than deleting everything in the workspace, but may not be applicable<br/>
for all situations.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dma_k">dma_k</a>:</i>
<p>The option to remove all workspace is important to make maven release plugin to<br/>
work correctly for the projects, that generate sources e.g. with JAXB. In my<br/>
case I commit JAXB sources from Windows environment, which after being<br/>
re-generated have different EOF and SVN senses this as uncommitted change, which<br/>
in its turn blocks the release.<br/>
If the fix is ready, what is the target milestone for this fix?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=smr99">smr99</a>:</i>
<p>What is the status of this?  I read through the referenced email thread (now at <a href="http://markmail.org/message/mcgftdni6cvpca7v" class="external-link" target="_blank" rel="nofollow noopener">http://markmail.org/message/mcgftdni6cvpca7v</a>) which seemed to end with agreement on the feature but issues with backwards compatibility.</p>
